cpp/src/copying/segmented_shift.cu

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/copying.hpp>
+#include <cudf/detail/copy.hpp>
+#include <cudf/detail/gather.cuh>
+#include <cudf/detail/iterator.cuh>
+#include <cudf/detail/scatter.hpp>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/types.hpp>
+
+#include <thrust/iterator/transform_iterator.h>
+
+namespace cudf {
+namespace detail {
+
+namespace {
+
+constexpr size_type SAFE_GATHER_IDX = 0;
```

Comment:
Hmnn, you should not use UPPER_CASE for variable name. UPPER_CASE should be used for enum only. Variable name is of snake_case, as stated here: https://github.com/rapidsai/cudf/blob/branch-0.19/cpp/docs/DEVELOPER_GUIDE.md#code-and-documentation-style-and-formatting